File: src/main/java/by/architecture_map/belarus/service/SourceService.java
package by.architecture_map.belarus.service;

import by.architecture_map.belarus.entity.Source;

import java.util.List;

public interface SourceService {

    Source create(Source source);

    Source find(int id);

    List<Source> findAll();

    Source update(int id, Source updatedSource);

    Source patchUpdate(int id, Source updatedSource);

    void delete(int id);
}

File: src/main/kotlin/by/architecture_map/belarus/service/SourceService.kt
package by.architecture_map.belarus.service

import by.architecture_map.belarus.entity.Source

interface SourceService {

    fun create(source: Source): Source
    fun find(id: Int): Source
    fun findAll(): List<Source>
    fun update(id: Int, updatedSource: Source): Source
    fun patchUpdate(id: Int, updatedSource: Source): Source
    fun delete(id: Int)
}
